### PR TITLE
[#2213] instagram UI alternative version

### LINKED ISSUE #2213: Instagram UI: Alternative Version
We have already made it possible to connect Instagram via the API but we should give system users the option to connect via the graphical UI also.

Option A: Connect an Instagram Account with Facebook Login -- Facebook Login Button
Option B: Connect an Instagram Account manually

This alternative version of connecting the Instagram UI should be available via a button above the `more information about this source` button. This additional button should allow users to connect their account with Facebook Login.

screenshot_2021-07-26_at_18 13 45

screenshot_2021-07-26_at_17 47 49

### Design Note
Once the system user clicks `Login with Facebook` we will inform system users:

To get started with Instagram Messages you need to make sure that:
- You have an Instagram Business Account ([documentation: how to switch to a Business account](https://www.facebook.com/help/instagram/502981923235522))
- Your Instagram Business Account needs to have at least 1,000 followers
- Your Instagram Business profile needs to be a connected to a Facebook page ([documentation](https://help.instagram.com/399237934150902?fbclid=IwAR1ekAUj69Zz9v6LduTdGbirlLjFcnTQv73TaVaC01tNi2iZtrkwsSlrhms)). We should make a note that linking can only be done via the Instagram mobile app.
- You need to allow API Access in Settings > Privacy > Messages in the Instagram App ([documentation](https://www.facebook.com/help/instagram/791161338412168))

### LINKED ISSUE #2616: Facebook Login for Facebook Messenger
related to #2213 
- Implement Facebook Login flow for Facebook Messenger

implemented in PR #2616

### PR DESCRIPTION
**Pull request by AudreyKj**
closes #2213 
closes #2616

**Facebook Login flow for Instagram and FB Messenger** 
- created a` FB Login` reusable component
- dropdown select if the user has multiple pages to connect
- added custom error message 
- fixed Instagram docs 
- small UI fixes

Login with Facebook button
Screenshot 2021-11-26 at 15 29 33

Once the user completed the login process using the button it displays "Logged In with Facebook"
Screenshot 2021-11-29 at 11 26 18

Screenshot 2021-12-02 at 10 11 42

### CODE REVIEW

File: docs/docs/sources/instagram.md [hunk 1]
```diff
@@ -36,9 +36,11 @@ Core Platform instance.
...
-- An instagram account fulfilling Facebook's [rollout criteria](https://developers.facebook.com/docs/messenger-platform/instagram/rollout)
-- A Facebook Page connected to that account
-- A registered [Facebook app](https://developers.facebook.com/docs/apps#register)
+- An instagram account fulfilling Facebook's [rollout criteria](https://developers.facebook.com/docs/messenger-platform/instagram/rollout).
+- A Facebook Page connected to that account (linking can only be done via the Instagram mobile app).
+- A registered [Facebook app](https://developers.facebook.com/docs/apps#register).
+
+You also need to allow API Access in Settings > Privacy > Messages in the Instagram mobile app.
 
 :::
```
Comment (AudreyKj) [inline docs/docs/sources/instagram.md:45]: Docs edited to make the requirements clearer.

File: frontend/ui/src/pages/Channels/FacebookLogin/index.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    <>
+      <button className={styles.facebookLogin} style={{cursor: isLoggedin ? 'not-allowed' : 'pointer'}} onClick={login}>
+        <FbLoginIcon />
+        <span>{isLoggedin ? 'Logged In' : 'Log In'} With Facebook</span>
+      </button>
+
+      {authError && <ErrorNotice theme="warning"> {authError} </ErrorNotice>}
+    </>
+  );
+};
```
Comment (AudreyKj) [inline frontend/ui/src/pages/Channels/FacebookLogin/index.tsx:93]: Re-usable `FacebookLogin` component that is both used for Instagram and Facebook.

File: frontend/ui/src/components/Sidebar/index.module.scss [hunk 3]
```diff
@@ -8,7 +8,6 @@
   border-top-right-radius: 8px;
   overflow-y: hidden;
   margin-top: 88px;
-  z-index: $navigation;
   width: 80px;
   height: auto;
   background-color: white;
```
Comment (AudreyKj) [inline frontend/ui/src/components/Sidebar/index.module.scss:13]: UI fix

File: frontend/ui/src/pages/Channels/ConnectedChannelsList/index.module.scss [hunk 4]
```diff
@@ -19,12 +19,15 @@
...
+.backButton:visited {
   display: block;
   margin-bottom: 16px;
   cursor: pointer;
   text-decoration: none;
   max-width: 200px;
+  color: var(--color-airy-blue);
   &:hover {
     text-decoration: underline;
   }
```
Comment (AudreyKj) [inline frontend/ui/src/pages/Channels/ConnectedChannelsList/index.module.scss:33]: UI fix

File: frontend/ui/src/pages/Channels/FacebookLogin/index.tsx (hunk 2 above)
Comment (AudreyKj) [reply frontend/ui/src/pages/Channels/FacebookLogin/index.tsx:138]: Screenshot 2021-12-02 at 10 11 42

File: frontend/ui/src/pages/Inbox/TemplateSelector/index.tsx [hunk 5]
```diff
@@ -91,7 +91,7 @@ const TemplateSelector = ({listTemplates, onClose, templates, selectTemplate, so
     <ListenOutsideClick onOuterClick={onClose}>
       <div className={styles.container} ref={componentRef}>
         {listTemplatesError ? (
-          <ErrorNotice theme="error">Oops! Your templates could not be loaded. Please try again later.</ErrorNotice>
+          <Notice theme="error">Oops! Your templates could not be loaded. Please try again later.</Notice>
         ) : templates.length === 0 && source === templatesSource ? (
           renderEmpty()
         ) : (
```
Comment (AudreyKj) [inline frontend/ui/src/pages/Inbox/TemplateSelector/index.tsx:97]: Refactored the `ErrorNotice `components and added the variant` info`.

File: lib/typescript/components/alerts/Notice/index.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+          <ExclamationIcon aria-hidden="true" />
+        </div>
+      )}
+
+      {children}
+    </div>
+  );
+};
+
+export const Notice = NoticeComponent;
```
Comment (AudreyKj) [inline lib/typescript/components/alerts/Notice/index.tsx:30]: Refactored the `ErrorNotice` components: added the variant `info` and renamed it to `Notice`.

File: docs/docs/sources/instagram.md (hunk 1 above)
Comment (AAnas222222) [reply docs/docs/sources/instagram.md:45]: Facebook

**Pull request closed without merging**